src/Fetch_Abort.res

```diff
@@ -1,10 +1,49 @@
+module Event = {
+  type t
+}
+
 module Signal = {
   type t
 
+  /**
+  https://dom.spec.whatwg.org/#dom-abortsignal-any
+  this is new as of 2024
+   */
+  @scope("AbortSignal")
+  external any: array<t> => t = "any"
+
+  @scope("AbortSignal")
+  external timeout: int => t = "timeout"
+
+  @scope("AbortSignal")
+  external abort: (~reason: 'reason=?) => t = "abort"
+
   @get external aborted: t => bool = "aborted"
-  @get @return(nullable) external reason: t => option<_> = "aborted"
+  @get @return(nullable) external reason: t => option<'reason> = "aborted"
 
   @send external throwIfAborted: t => unit = "throwIfAborted"
+
+  @set external onabort: (t, Event.t => unit) => unit = "onabort"
+
+  // inherited from EventTarget
+  type addEventListenerOptions = {
+    capture?: bool,
+    once?: bool,
+    passive?: bool,
+    signal?: t,
+  }
+
+  @send
+  external addEventListener: (t, @string [#abort(Event.t => unit)], ~options: addEventListenerOptions=?) => unit =
```

Comment:
@glennsl what do you think of forcing the event name argument to be `"abort"` like this:

```rescript
@send
external addEventListener: (t, @as("abort") _, Event.t => unit, ~options: addEventListenerOptions=?) => unit = "addEventListener"
```

So that users don't have to wrap the event listener with the `#abort(...)` variant?

Reply:
I think it's good as is, since it seems more consistent with what users would expect from using `addEventListener` with other objects that derive from `EventTarget`.